gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcClientCalls.java

```diff
@@ -57,6 +58,11 @@
     CallOptions callOptions = grpcContext.getCallOptions();
     Preconditions.checkNotNull(callOptions);
 
-    return grpcContext.getChannel().newCall(descriptor, callOptions);
+    Channel channel = grpcContext.getChannel();
+    if (grpcContext.getChannelAffinity() != null && channel instanceof ChannelPool) {
```

Comment:
The `instanceof` usually indicates that there is something wrong with OOP design of the classes. Is it possible for the `grpcContext.getChannel()` itself already return a proper channel?

Reply:
It is, but then we have to move the `instanceof` check into `GrpcCallContext` right?

I think if we want to "do it right" the OOP way, we have to stop using the plain `Channel` abstraction and pass around `ChannelPool` instead, where a single `Channel` must be converted into a pool of one. I think this is too disruptive though.

Another way, we can define
```java
interface ChannelWithAffinity {
  Channel getWithAffinity(int affinity);
}
```
and check `instanceof ChannelWithAffinity` instead similar to how `ArrayList` implements `RandomAccess`.

I don't particularly like any of the above options, so I went with the current code because it takes the least work :laughing: and easiest to change later.

Reply:
I don't think there is a super easy and lightweight solution to this problem. I agree that hiding `Channel` and `ChannelPool` under same abstraction would bee too disruptive. I've just realized that this class (`GrpcClientCalls`) has only one method and it already has an explicit type conversion (L55, so type shenanigans are already in the spirit of this method). Ok, lets keep it as is for now.